core/src/test/scalajvm/sttp/client/testing/websocket/HighLevelWebsocketTest.scala

```diff
@@ -67,6 +71,20 @@ abstract class HighLevelWebsocketTest[F[_], WS_HANDLER[_]]
       .toFuture()
   }
 
+  it should "fail correctly when can't open WebSocket" in {
+    implicit val signaler: Signaler = ThreadSignaler
+    failAfter(Span(15, Seconds)) {
```

Comment:
what happens when there's no time limit? it hangs? and why 15 seconds, seems a lot ;)

Reply:
It hangs without fix added in this PR.
So if there is such bug again, perhaps in other backend or some regression, this test will fail after 15 seconds.
If implementation is correct this test passes immediately.
15 seconds seems safe for CI.

Reply:
👍